src/Sarif.Viewer.VisualStudio.UnitTests/ErrorList/SarifFileWithContentsTests.cs

```diff
@@ -97,35 +181,79 @@ public void SarifFileWithContents_SavesContents()
         {
             var fileDetails = CodeAnalysisResultManager.Instance.FileDetails;
 
-            fileDetails.Should().ContainKey(Key);
+            fileDetails.Should().ContainKey(key1);
         }
 
         [Fact]
-        public void SarifFileWithContents_DecodesContents()
+        public void SarifFileWithContents_DecodesBinaryContents()
         {
-            var fileDetail = CodeAnalysisResultManager.Instance.FileDetails[Key];
+            var fileDetail = CodeAnalysisResultManager.Instance.FileDetails[key2];
             var contents = fileDetail.GetContents();
 
-            contents.Should().Be(Contents);
+            contents.Should().Be(expectedContents2);
         }
 
         [Fact]
-        public void SarifFileWithContents_OpensEmbeddedFile()
+        public void SarifFileWithContents_OpensEmbeddedBinaryFile()
         {
-            var rebaselinedFile = CodeAnalysisResultManager.Instance.CreateFileFromContents(Key);
-
+            var rebaselinedFile = CodeAnalysisResultManager.Instance.CreateFileFromContents(key2);
             var fileText = File.ReadAllText(rebaselinedFile);
 
-            fileText.Should().Be(Contents);
+            fileText.Should().Be(expectedContents2);
+        }
+
+        [Fact]
+        public void SarifFileWithContents_DecodesTextContents()
+        {
+            var fileDetail = CodeAnalysisResultManager.Instance.FileDetails[key3];
+            var contents = fileDetail.GetContents();
+
+            contents.Should().Be(expectedContents1);
```

Comment:
These tests don't verify the hashing code. For each one, verify the file hash value.